grpc-bridge/src/main/java/org/jboss/resteasy/grpc/protobuf/JavaToProtobufGenerator.java

```diff
@@ -0,0 +1,1032 @@
+      HTTP_VERBS.add("GET");
+      HTTP_VERBS.add("OPTIONS");
+      HTTP_VERBS.add("PATCH");
+      HTTP_VERBS.add("POST");
+      HTTP_VERBS.add("PUT");
+
+      SSE_EVENT = "package org.jboss.resteasy.plugins.grpc.sse;\n"
+                + "\n"
+                + "public class SseEvent {\n"
+                + "\n"
+                + "   String  comment;\n"
+                + "   String  id;\n"
+                + "   String  name;\n"
+                + "   byte[]  data;\n"
+                + "   long    reconnectDelay;\n"
+                + "}";
+   }
+
+   public static void main(String[] args) throws IOException
+   {
+      if (args == null || (args.length != 4)) {
+         logger.info("need four args");
+         logger.info("  arg[0]: root directory");
+         logger.info("  arg[1]: package to be used in .proto file");
+         logger.info("  arg[2]: java package to be used in .proto file");
+         logger.info("  arg[3]: java outer classname to be generated from .proto file");
+         logger.info("  -Djars: comma separated of jars [optional]");
+         logger.info("  -Dclasses: comma separated of addition classes [optional]");
+         return;
+      }
+      prefix = args[3];
+      String s = System.getProperty("jars", "default");
+      jars = "default".equals(s) || "".equals(s)
+            ? new CopyOnWriteArraySet<String>()
+            : new CopyOnWriteArraySet<String>(Arrays.asList(s.split(",")));
+      s = System.getProperty("classes", "default");
+      additionalClasses = "default".equals(s) || "".equals(s)
+            ? new CopyOnWriteArraySet<String>()
+            : new CopyOnWriteArraySet<String>(Arrays.asList(s.split(",")));
+      StringBuilder sb = new StringBuilder();
+      protobufHeader(args, sb);
+      new JavaToProtobufGenerator().processClasses(args, sb);
+      while (!resolvedTypes.isEmpty()) {
+         for (ResolvedReferenceTypeDeclaration rrtd : resolvedTypes) {
+            classVisitor.visit(rrtd, sb);
+         }
+      }
+      finishProto(sb);
+      writeProtoFile(args, sb);
+      createProtobufDirectory(args);
+   }
+
+   private static void protobufHeader(String[] args, StringBuilder sb)
+   {
+      sb.append("syntax = \"proto3\";\n");
+      sb.append("package " + args[1].replace('-', '.') + ";\n");
+      sb.append("import \"google/protobuf/any.proto\";\n");
+      sb.append("import \"google/protobuf/timestamp.proto\";\n");
+      sb.append("option java_package = \"" + args[2] + "\";\n");
+      sb.append("option java_outer_classname = \"" + args[3] + "_proto\";\n");
+   }
+
+   /**
+    * Visit all JAX-RS resource classes discovered in project's src/main/java
+    */
+   private void processClasses(String[] args, StringBuilder sb) throws IOException {
+      Log.setAdapter(new Log.StandardOutStandardErrorAdapter());
+
+      // SourceRoot is a tool that read and writes Java files from packages on a certain root directory.
+      Path path = Path.of(args[0], "/src/main/java/");
```

Comment:
Something to consider is this will only work with a standard maven build with a path like this.

Reply:
True. I'll work on it later.